src/bootstrap/test.rs

```diff
@@ -475,6 +475,8 @@ impl Step for Miri {
         let stage = self.stage;
         let host = self.host;
         let compiler = builder.compiler(stage, host);
+        // Get a "at least stage 1" compiler, that is suitable for building the stdlib for us.
+        let compiler_std = builder.compiler(std::cmp::max(stage, 1), host);
```

Comment:
@jyn514 remarked that this should probably be +1, not `max`. However, `--stage 1` works right now, we just need to overcome the bootstrap loop -- once we are at stage 1, we have a compiler and stdlib built with the same compiler, so we no longer need to do anything extra here.

Reply:
I now made this `stage+1` for `stage < 2`, to overcome potential issues on proc macro ABI changes.